scripts/T6_objective_api_main.gsc

```diff
@@ -37,11 +37,7 @@ on_player_disconnect()
         hud_keys = getArrayKeys( level.custom_objectives );
         foreach ( key in hud_keys )
         {
-            player_elem_keys = getArrayKeys( level.custom_objectives[ key ].entities );
-            for ( j = 0; j < player_elem_keys.size; j++ )
-            {
-                level.custom_objectives[ key ].entities[ player_elem_keys[ j ] ] notify( "destroy_hud_ent" );
-            }
+            level.custom_objectives[ key ].entities[ self.name ] notify( "destroy_hud_ent" );
```

Comment:
this is unreliable code and, if changed, will break. maybe do a GUID/XUID that doesn't change so easily (i know its changeable but no one ever changes that tbh)